solr/core/src/java/org/apache/solr/cli/HealthcheckTool.java

```diff
@@ -60,11 +60,11 @@ public List<Option> getOptions() {
         SolrCLI.OPTION_SOLRURL,
         SolrCLI.OPTION_ZKHOST,
         Option.builder("c")
-            .argName("COLLECTION")
+            .longOpt("name")
```

Comment:
[Q] I've picked HealthcheckTool's handling of its "collection" parameter as a concrete example, but the concern below applies in a number of places.

----

Historically, `bin/solr healthcheck` has accepted the collection name using either `-c` or `-collection`.  But with this PR, AFAICT it now accepts `-c` and `--collection` (note the double-dash in the longopt).

Personally, I like the new syntax better - double-dashes on longopts are pretty ubiquitous in other CLI tools, and Solr's usage of `-longoptName` always struck me as quirky.  That said though, it's a backwards incompatible change so we need to roll it out carefully.

If you **do** plan on committing some form of this to branch_9x, do you have plans for addressing backcompat?  (i.e. Would the branch_9x version accept both `--collection` and `-collection` as longopt forms?) 

If you **don't** plan on committing some form of this to branch_9x, things are better from a backcompat perspective but we'd still need to call out the syntax changes in CHANGES.txt or our Upgrade Notes, presumably?Do you have a general sense or (even better) a list of the syntax changes that'd need documented?

Reply:
Thanks for jumping on this...   https://issues.apache.org/jira/browse/SOLR-16824 was opened to specifically track that conversion to the `--collection` approach.   I agree on the syntax changes needing tracking in Upgrade Notes.   I might start with the Upgrade Notes for 10.x, but if we back port, then bring them back to that version of Upgrade Notes.

Reply:
also...   one more thing is that the long name is actually `--name`..   I was struggling with the CreateTool which used `--name` since it can do core or collection.   Can't use `--collection` for a Core!

Reply:
LIkewise the same thinking with the DeleteTool